aten/src/ATen/native/DistributionTemplates.h

```diff
@@ -235,8 +199,10 @@ Tensor& normal_out_impl(Tensor& output, const Tensor& mean, double std, c10::opt
 template<template<typename> class normal_kernel, typename RNG>
 Tensor& normal_out_impl(Tensor& output, double mean, const Tensor& std, c10::optional<Generator> gen) {
   CHECK_NORMAL_TENSOR_STD(std);
-  normal_impl_<normal_kernel, RNG>(output, 0, 1, gen);
   auto mean_tensor = at::full({}, mean, output.options());
+  auto shape = at::infer_size(mean_tensor.sizes(), std.sizes());
+  at::native::resize_output(output, shape);
```

Comment:
I might be wrong - do we actually need to change this function?

This looks like a behavior change. If I run this on master:
```
>>> import torch
>>> output = torch.tensor([])
>>> std = torch.ones(4)
>>> torch.normal(3, std)
tensor([2.2288, 2.9681, 2.4274, 2.2930])
>>> torch.normal(3, std, out=output)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: The size of tensor a (0) must match the size of tensor b (4) at non-singleton dimension 0
```

So it looks like today, passing a size=0 `out=` tensor will raise an error instead of resizing the tensor. We probably want to keep that behavior.

Reply:
i don't think the current behavior is correct because in the future out with size 0 will be resized, no? so this is the step in the right direction imo. but feel free to clarify

Reply:
ok yep, agreed. When I was originally reviewing the PR I was leaning towards "try to avoid making functionality changes to the operator when we add a meta API". But you're right, this seems like strictly better behavior (we used to incorrectly error, now we do the right thing).